### PR TITLE
EZP-24471: create a test suite for Solr search engine multicore implementation

### PR DESCRIPTION
**Pull request by pspanja**
> Review dependency graph
> * https://github.com/ezsystems/ezpublish-kernel/pull/1305 Main languages core
>  * **https://github.com/ezsystems/ezpublish-kernel/pull/1322 Multicore tests**
>    * https://github.com/ezsystems/ezpublish-kernel/pull/1326 Complete CustomField support
>       * https://github.com/ezsystems/ezpublish-kernel/pull/1327 Extract CoreFilter service
>       * https://github.com/ezsystems/ezpublish-kernel/pull/1333 Indexable Selection
>          * https://github.com/ezsystems/ezpublish-kernel/pull/1334 Indexable RelationList
> * https://github.com/ezsystems/ezpublish-kernel/pull/1329 DateModified sort clause

#### Based on https://github.com/ezsystems/ezpublish-kernel/pull/1305

This implements unit tests for endpoint resolution and integration tests for language fallback in multicore Solr search engine.
Previous tests were only adapted from the tests for the different behavior in Legacy and Elasticsearch engines and are therefore here removed.

Implementing the tests required some changes in the API namespace:

1. `SearchHit` received `$contentTranslation` property, holding the language code of the translation that the query matched. In future that information can be used in the `SearchService` to load only that single translation, which should be the one displayed on the site front.
2. `SearchService::findLocations()` received `$fieldFilters` parameter. As Location Search can now search and sort on Content field data, the parameter is needed to provide proper language fallback. Fulltext search was previously disabled.

  :exclamation: Note that the parameter was added as the second parameter, moving `$filterOnUserPermissions` parameter to the 3rd place. That is a bit of a BC break, but it is in line with `findContent()` method and will be done for a major release.

Tests assert correct result set, including originating index (`SearchHit::$index`, shard in Solr speak) and matched translation, for 3 different setups:

1. **multicore dedicated**

  uses a separate core for each language, no default and main languages cores

2. **multicore shared**

  uses default and main languages cores together with mapping a dedicated core for a language, also mixes cores for both Content and Location documents

3. **single core**

  uses single core for all translations, for both Content and Location data

Tests are implemented in a separate test case. Main `SearchService` test case will set `$index` and `$contentTranslation` properties in `SearchHit` to `null`, as these tests are common for all engines and the properties are populated only by Solr engine ATM.

#### Additionally fixed

Default value for `$fieldFilters["useAlwaysAvailable"]` was false, which is incorrect towards the documentation and implementation for Content loading.

### CODE REVIEW

**Comment by pspanja** [comment]
Ready for the review, ping @andrerom @paulborgermans @lolautruche @bdunogier

**Comment by bdunogier** [comment]
I guess that this PR isn't meant to be backported ?

**Comment by pspanja** [comment]
I should be backported, but without parameter order change on `SearchService::findLocations()`, so `$fieldFilters` should be on 3rd place. What is your POV on that, should we rather do the same here as well?

**Comment by bdunogier** [comment]
It is 100% clear that the API change can not be backported, yes. 

About 1.x and above, I'm not sure. I'd rather lean towards making it more consistent...

**Comment by pspanja** [comment]
That's my preference as well obviously :)
Now Travis gave up and passed the tests, but this depends on #1305, which should be merged first.

File: .travis.yml [hunk 1]
```diff
@@ -52,6 +52,7 @@ matrix:
 branches:
   only:
     - master
+    - impl-EZP-24434-aa-core-2
```
Comment (andrerom) [inline .travis.yml]: remember to remove

**Comment by andrerom** [comment]
+1, and the API change is ok, from my POV we should aim to remove filterOnUserPermissions eventually.

File: .travis.yml (hunk 1 above)
Comment (pspanja) [inline .travis.yml]: Removed in https://github.com/ezsystems/ezpublish-kernel/commit/c7eaa711d0f3afaa1ee2258951528e9f821c15b8

**Comment by pspanja** [comment]
Manually merged in https://github.com/ezsystems/ezpublish-kernel/commit/c7eaa711d0f3afaa1ee2258951528e9f821c15b8

**Pull request closed without merging**